server/src/main/java/io/unitycatalog/server/persist/StagingTableRepository.java

```diff
@@ -0,0 +1,204 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.CreateStagingTable;
+import io.unitycatalog.server.model.StagingTableInfo;
+import io.unitycatalog.server.persist.dao.StagingTableDAO;
+import io.unitycatalog.server.persist.dao.TableInfoDAO;
+import io.unitycatalog.server.persist.utils.RepositoryUtils;
+import io.unitycatalog.server.persist.utils.TransactionManager;
+import io.unitycatalog.server.utils.IdentityUtils;
+import io.unitycatalog.server.utils.ServerProperties;
+import io.unitycatalog.server.utils.ValidationUtils;
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.commons.lang3.tuple.Pair;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StagingTableRepository {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StagingTableRepository.class);
+  private final SessionFactory sessionFactory;
+  private final Repositories repositories;
+  private final ServerProperties serverProperties;
+
+  public StagingTableRepository(
+      Repositories repositories, SessionFactory sessionFactory, ServerProperties serverProperties) {
+    this.repositories = repositories;
+    this.sessionFactory = sessionFactory;
+    this.serverProperties = serverProperties;
+  }
+
+  /**
+   * Retrieves a staging table by its unique identifier.
+   *
+   * @param stagingTableId the UUID of the staging table to retrieve
+   * @return StagingTableInfo containing the staging table details
+   * @throws BaseException with ErrorCode.NOT_FOUND if the staging table does not exist
+   */
+  public StagingTableInfo getStagingTableById(String stagingTableId) {
```

Comment:
Could you please remove this in this PR, if we neither have any usage for it or test for it ? Because it seems unrelated to the changes.

If it's for the next PR, then it will be better to introduce in the next PR. since we can have test for that.